govcd/edgegateway.go

```diff
@@ -665,7 +665,7 @@ func (e *EdgeGateway) AddIpsecVPN(ipsecVPNConfig *types.EdgeGatewayServiceConfig
 
 	output, err := xml.MarshalIndent(ipsecVPNConfig, "  ", "    ")
 	if err != nil {
-		fmt.Errorf("error marshaling ipsecVPNConfig compose: %s", err)
+		return fmt.Errorf("error marshaling ipsecVPNConfig compose: %s", err)
```

Comment:
This will result in a compilation error. The function returns a Task and an error

Reply:
@auppunda I have updated the PR